locals.tf

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  keyring_id   = var.existing_keyring ? data.google_kms_key_ring.key_ring[0].id : google_kms_key_ring.key_ring[0].id
+  key_opts_map = { for opts in var.key_opts : opts.key => opts }
+
+  key_opts_default = {
+    rotation_period  = "86400s"
```

Comment:
Seems like we are changing existing default?https://github.com/terraform-google-modules/terraform-google-kms/blob/c28453ccb6ed33e2c9c7a4e610b2775424536d50/variables.tf#L86

Reply:
yep, just a more sane default, the previous one was just copied over from the tf docs. Though if you think it's worth keeping, lets change it back?

Reply:
I think 1 day is reasonable, lets keep this as default.